File: app/src/main/java/com/pawels96/skyrimperkcalculator/enums/IPerk.java
package com.pawels96.skyrimperkcalculator.enums;

import com.pawels96.skyrimperkcalculator.models.PerkInfo;

import java.io.Serializable;

/**
 * Interface implemented by all perk enums.
 * Returns an object which contains X and Y coordinates of a perk and required skill levels.
 */

public interface IPerk extends Serializable {
   PerkInfo getPerkInfo();
}

File: app/src/main/java/com/pawels96/skyrimperkcalculator/domain/IPerk.kt
package com.pawels96.skyrimperkcalculator.domain

import java.io.Serializable

/**
 * Interface implemented by all perk enums
 */
interface IPerk : Serializable {

    val perkInfo: PerkInfo

    val childPerks: List<IPerk>
}

/**
 * Implemented by perk enums belonging to skills with non-standard requirements for perks: Vampirism and Lycanthropy
 */
interface SpecialSkillPerk : IPerk
